hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4949,6 +4950,25 @@ func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx c
 		}
 	}
 
+	// Reconcile management infrastructure annotation
+	mgmtInfraKey := client.ObjectKey{Name: "cluster"}
+	mgmtInfra := &configv1.Infrastructure{}
+
+	if err := r.Get(ctx, mgmtInfraKey, mgmtInfra); err != nil {
+		return fmt.Errorf("failed to get infrastructure.config.openshift.io status: %w", err)
+	}
+	mgmtPlatformType := mgmtInfra.Status.PlatformStatus.Type
```

Comment:
why not move this section into the if statement that checks for the mgmt annotation.

```
if _, exists := hc.Annotations[hyperv1.ManagementPlatformAnnotation]; !exists {
```

That way, we're only doing a Get() on the infrastructure resource when the mgmt annotation doesn't already exist on the hc.

Reply:
done.